### PR TITLE
Use class property transforms and actions from props

### PR DESCRIPTION
**Pull request by rueckstiess**
this is nice: https://babeljs.io/docs/plugins/transform-class-properties/

and passing actions into props so components are pure and nicely testable.

### CODE REVIEW

File: template/package.json [hunk 1]
```diff
@@ -9,7 +9,7 @@
     "fmt": "mongodb-js-fmt ./*.js ./test/*.js",
     "check": "mongodb-js-precommit './src/**/*{.js,.jsx}' './test/**/*.js' index.js",
     "compile": "npm run jsx && npm run less",
-    "jsx": "babel ./src --out-dir ./lib --presets react",
+    "jsx": "babel --presets react --plugins transform-class-properties --out-dir ./lib ./src",
```
Comment (hswolff) [inline template/package.json]: Does babel not use the babelrc file by default? Is there an argument to set the config file to use? It'd be nice to not repeat configuration.

File: template/src/components/index.jsx [hunk 2]
```diff
@@ -7,6 +7,9 @@ const Actions = require('../actions');
 // const debug = require('debug')('mongodb-compass:{{slugcase name}}:index');
 
 class Connected{{pascalcase name}}Component extends React.Component {
+
+  static displayName = 'Connected{{pascalcase name}}Component'
```
Comment (hswolff) [inline template/src/components/index.jsx]: I thought this was handled by default and didn't need to be set explicitly.

File: template/src/components/{{name}}.jsx [hunk 3]
```diff
@@ -1,14 +1,23 @@
...
+
+  static propTypes = {
+    status: PropTypes.oneOf(['enabled', 'disabled'])
+  }
+
+  static defaultProps = {
+    status: 'enabled'
+  }
+
+  onToggle = () => {
```
Comment (hswolff) [inline template/src/components/{{name}}.jsx]: Wohoo!

File: template/package.json (hunk 1 above)
Comment (rueckstiess) [inline template/package.json]: oh it does, too. even better!

**Comment by rueckstiess** [comment]
Superseeded by the recent webpack changes, which will include class property support

**Pull request closed without merging**